src/Runner.Worker/ActionManager.cs

```diff
@@ -51,6 +52,168 @@ public sealed class ActionManager : RunnerService, IActionManager
         private readonly Dictionary<Guid, ContainerInfo> _cachedActionContainers = new Dictionary<Guid, ContainerInfo>();
 
         public Dictionary<Guid, ContainerInfo> CachedActionContainers => _cachedActionContainers;
+        public async Task<PrepareResult> PrepareActionsV2Async(IExecutionContext executionContext, IEnumerable<Pipelines.JobStep> steps)
```

Comment:
notes for reader: this is the download code

Reply:
let's check whether legacy can be deleted (depends when it shipped on GHES)

Also might be interesting to get a good answer how far back we support (latest runner, old GHES)

Reply:
rather than fork the code, have you considered checking using the new feature flag to error/allow nested composite references? forking the code seems less desirable if the feature flag code will not be removed any time soon